crates/sozo/ops/src/migration/mod.rs

```diff
@@ -683,3 +692,78 @@ pub fn handle_artifact_error(ui: &Ui, artifact_path: &Path, error: anyhow::Error
          --artifacts` to clean artifacts only.\nThen, rebuild your project with `sozo build`."
     )
 }
+
+pub async fn get_contract_operation_name<P>(
+    provider: &P,
+    contract: &ContractMigration,
+    world_address: Option<FieldElement>,
+) -> String
+where
+    P: Provider + Sync + Send + 'static,
+{
+    if let Some(world_address) = world_address {
+        if let Ok(base_class_hash) = provider
+            .call(
+                FunctionCall {
+                    contract_address: world_address,
+                    calldata: vec![],
+                    entry_point_selector: get_selector_from_name("base").unwrap(),
+                },
+                BlockId::Tag(BlockTag::Pending),
+            )
+            .await
+        {
```

Comment:
@lambda-0x just to keep a note aside, after the modification on upgrade for the world + outputting the base class hash in the manifest, we shouldn't fetch, but only take the base class hash from the local manifest. :+1: